fractions/microprofile/microprofile-jwt/src/main/java/org/wildfly/swarm/microprofile/jwtauth/runtime/MPJWTAuthExtensionArchivePreparer.java

```diff
@@ -143,9 +146,21 @@ public void process() throws Exception {
             log.debugf("Issuer: %s", fraction.getTokenIssuer().get());
             war.addAsManifestResource(new StringAsset(fraction.getTokenIssuer().get()), "MP-JWT-ISSUER");
         }
-        if (fraction.getPublicKey() != null) {
-            log.debugf("PublicKey: %s", fraction.getPublicKey());
-            war.addAsManifestResource(new StringAsset(fraction.getPublicKey()), "MP-JWT-SIGNER");
+
+        String publicKey = fraction.getPublicKey();
+        if (publicKey != null) {
+            log.debugf("PublicKey: %s", publicKey);
+
+            if (publicKey.startsWith("file:")) {
+                File fileRef = new File(publicKey.substring(5, publicKey.length()));
+                war.addAsManifestResource(new FileAsset(fileRef), "MP-JWT-SIGNER");
+            } else if (publicKey.startsWith("classpath:")) {
+                String cpref  = publicKey.substring(10, publicKey.length());
+                Node node = archive.get("WEB-INF/classes/" + cpref);
```

Comment:
Is `WEB-INF/classes` the only place? What about a `.jar`?

Reply:
hm, good point

Reply:
Ah I forgot this is all based around web. Good point.